src/android.ts

```diff
@@ -237,6 +238,121 @@ export class AndroidRobot implements Robot {
 
 		return rotation === "0" ? "portrait" : "landscape";
 	}
+
+	private getParsedXml(): UiAutomatorXml {
+		const dump = this.adb("exec-out", "uiautomator", "dump", "/dev/tty");
+
+		const parser = new xml.XMLParser({
+			ignoreAttributes: false,
+			attributeNamePrefix: ""
+		});
+
+		return parser.parse(dump) as UiAutomatorXml;
+	}
+
+	private getScreenElementRect(node: UiAutomatorXmlNode): ScreenElementRect {
+		const bounds = String(node.bounds);
+
+		const [, left, top, right, bottom] = bounds.match(/^\[(\d+),(\d+)\]\[(\d+),(\d+)\]$/)?.map(Number) || [];
+		return {
+			x: left,
+			y: top,
+			width: right - left,
+			height: bottom - top,
+		};
+	}
+
+	// --- TV Specific Methods ---
+
+	private async tvTap(x: number, y: number): Promise<void> {
+		let currentDirection = this.getDpadDirection(x, y);
+
+		while (currentDirection) {
+			this.pressDpad(currentDirection);
+			currentDirection = this.getDpadDirection(x, y);
+		}
+
+		// Upon breaking from the while loop, we have reached at the target coordinates
+		// Press the dpad center button to 'click' on the focused element
+		this.pressDpad("DPAD_CENTER");
+	}
+
+	/**
+	 * Find the focused element in the UI hierarchy.
+	 *
+	 * @param node - The root node of the UI hierarchy.
+	 * @returns The focused element node or null if not found.
+	 */
+	private findFocusedElement(node: UiAutomatorXmlNode): UiAutomatorXmlNode | null {
+		if (node["focused"] === "true") {
+			return node;
+		}
+
+		if (node.node) {
+			if (Array.isArray(node.node)) {
+				for (const childNode of node.node) {
+					const focusedChild = this.findFocusedElement(childNode);
+					if (focusedChild) {
+						return focusedChild;
+					}
+				}
+			} else {
+				const focusedChild = this.findFocusedElement(node.node);
+				if (focusedChild) {
+					return focusedChild;
+				}
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Get the dpad direction based on the target coordinates.
+	 *
+	 * @param targetX - The target X coordinate.
+	 * @param targetY - The target Y coordinate.
+	 * @returns The dpad direction or null if no dpad direction is needed.
+	 */
+	private getDpadDirection(targetX: number, targetY: number): DpadButton | null {
+		// Always use the latest parsed XML. When dpad keyevent is sent, the screen is updated and
+		// the focused element may have changed.
+		const parsedXml = this.getParsedXml();
+		const focusedElement = this.findFocusedElement(parsedXml.hierarchy.node);
```

Comment:
`memo`
With the latest layout information, I know which is my currently focused element and which direction I need to be heading to in order to reach the target coordinates.

That is the reason why before searching for the focused element `findFocusedElement` I need to call `getParseXml` to get the latest layout information